src/d3a/models/market/two_sided.py

```diff
@@ -264,9 +280,40 @@ def match_recommendation(self, recommended_list):
                     recommended_list, index+1, trade, bid_trade
                 )
 
+    @staticmethod
+    def validate_requirements_satisfied(
+            offer: Offer, bid: Bid, clearing_rate: float = None,
+            selected_energy: float = None) -> None:
+        """Validate if both trade parties satisfy each other's requirements.
+
+        :raises:
+            InvalidBidOfferPairException: Bid offer pair failed the validation
+        """
+        if ((offer.requirements or bid.requirements) and
+                not RequirementsSatisfiedChecker.is_satisfied(
+                    offer, bid, clearing_rate, selected_energy)):
+            # If no requirement dict is satisfied
+            raise InvalidBidOfferPairException(
+                f"OFFER: {offer} & BID: {bid} requirements failed the validation.")
+
+    @classmethod
+    def validate_authentic_bid_offer_pair(
+            cls, bid: Bid, offer: Offer, clearing_rate: float, selected_energy: float) -> None:
+        """Basic validation function for bid against an offer.
+
+        :raises:
+            InvalidBidOfferPairException: Bid offer pair failed the validation
+        """
+        if not (bid.energy >= selected_energy and
+                offer.energy >= selected_energy and
+                (bid.energy_rate + FLOATING_POINT_TOLERANCE) >= clearing_rate and
+                (bid.energy_rate + FLOATING_POINT_TOLERANCE) >= offer.energy_rate):
+            raise InvalidBidOfferPairException
+        cls.validate_requirements_satisfied(bid, offer, clearing_rate, selected_energy)
```

Comment:
The arguments are in the wrong order here: you're passing the bid instead of the offer.
```suggestion
        cls.validate_requirements_satisfied(offer, bid, clearing_rate, selected_energy)
```
To make it clearer, please use named arguments.

BTW, this was spotted by pylint. Please make sure you run it again on all files before asking for reviews, so that we can proceed faster on this kind of warnings :)

Reply:
Also: if this didn't fail in the tests, it probably means that there's not enough coverage. It would be great if you could add tests for this part.

Reply:
I'm building the habit of doing that progressively, although sometimes I introduce a `small should not harm` changes without rerunning pylint 😄

Reply:
and FYI I'm adapting in the other PR as I  thought it could make it easier for me and the reviewer